From fc1b5a0085b4eae951154f6d79f3d02bafa8d5f4 Mon Sep 17 00:00:00 2001
From: Qingqing Zhou <quic_qqzhou@quicinc.com>
Date: Thu, 19 Sep 2024 12:26:24 +0530
Subject: [PATCH 19/68] iommu/arm-smmu: Workaround the SMMU_IDR2 value

On Talos, some smmu clients meet with smmu context fault
issue, the fault iovas are all 0xfffffe000, clients use
40 bits iova returned from dma_alloc_coherent, but it
seems smmu hardware only supports 36 bits width of iova
even though SMMU_IDR2 register shows 48-bit width support.

So workaround this to unblock smmu clients.

Signed-off-by: Qingqing Zhou <quic_qqzhou@quicinc.com>
---
 drivers/iommu/arm/arm-smmu/arm-smmu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/iommu/arm/arm-smmu/arm-smmu.c b/drivers/iommu/arm/arm-smmu/arm-smmu.c
index 8321962b3714..e9d997c963c6 100644
--- a/drivers/iommu/arm/arm-smmu/arm-smmu.c
+++ b/drivers/iommu/arm/arm-smmu/arm-smmu.c
@@ -1877,6 +1877,8 @@ static int arm_smmu_device_cfg_probe(struct arm_smmu_device *smmu)
 
 	/* ID2 */
 	id = arm_smmu_gr0_read(smmu, ARM_SMMU_GR0_ID2);
+	dev_err(smmu->dev, "SMMU_ID2=0x%x\n", id);
+	id = 0x5111;
 	size = arm_smmu_id_size_to_bits(FIELD_GET(ARM_SMMU_ID2_IAS, id));
 	smmu->ipa_size = size;
 
-- 
2.34.1

